cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1704,7 +1714,7 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         }
 
         Future<ReconciliationState> entityOperatorDeployment(Supplier<Date> dateSupplier) {
-            if (this.entityOperator != null) {
+            if (this.entityOperator != null && eoDeployment != null) {
```

Comment:
tbh I didn't run the code, but it really possible that if `entityOperator` is not null, the `eoDeployment` could be null and we have to check here? I see that the `eoDeployment` generation code : 

```java
this.eoDeployment = entityOperator.generateDeployment(isOpenShift, annotations, imagePullPolicy);
```

is already inside the block where `entityOperator` is not null.

Reply:
Yes, I added this because of the exception which was thrown in this code. The `eoDeployment` will be null when neither TO nor UO are specified - i.e. when you do this:

```yaml
entityoperator: {}
```

In this case the `entityOperator` seems to be not null but the `eoDeployment` is. That is why I added the check.